thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/monitor/MonitorTaskRunner.java

```diff
@@ -113,11 +117,35 @@ private void executeMonitorUpdate(MonitorTaskInfo monitorTaskInfo) {
           LOG.info("COMPLETED jobs {}", scheduledJobs.keySet());
         }
       }
+
+      // update detection health
+      updateDetectionHealth();
     } catch (Exception e) {
       LOG.error("Exception in monitor update task", e);
     }
   }
 
+  private void updateDetectionHealth() {
+    DetectionConfigManager detectionDAO = DAO_REGISTRY.getDetectionConfigManager();
+    List<DetectionConfigDTO> detectionConfigs = detectionDAO.findAllActive();
+    for (DetectionConfigDTO config : detectionConfigs) {
+      // update detection health status
+      try {
+        DateTime healthStatusWindowEnd = DateTime.now();
+        config.setHealth(new DetectionHealth.Builder(config.getId(), healthStatusWindowEnd.minusDays(30).getMillis(),
+            healthStatusWindowEnd.getMillis()).addRegressionStatus(DAO_REGISTRY.getEvaluationManager())
+            .addAnomalyCoverageStatus(DAO_REGISTRY.getMergedAnomalyResultDAO())
+            .addDetectionTaskStatus(DAO_REGISTRY.getTaskDAO())
+            .addOverallHealth()
+            .build());
+        detectionDAO.update(config);
```

Comment:
How long does the previous build cost?
It may have racing-condition with other update modules.
Suggest to get the latest config again before actually updating it.

Reply:
Good point. I've updated it to read again before writing the config.